ext/random/random.stub.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/** @generate-class-entries */
+
+namespace
+{
+
+    function lcg_value(): float {}
+
+    function mt_srand(int $seed = 0, int $mode = MT_RAND_MT19937): void {}
+
+    /** @alias mt_srand */
+    function srand(int $seed = 0, int $mode = MT_RAND_MT19937): void {}
+
+    function rand(int $min = UNKNOWN, int $max = UNKNOWN): int {}
+
+    function mt_rand(int $min = UNKNOWN, int $max = UNKNOWN): int {}
+
+    function mt_getrandmax(): int {}
+
+    /** @alias mt_getrandmax */
+    function getrandmax(): int {}
+
+    /** @refcount 1 */
+    function random_bytes(int $length): string {}
+
+    function random_int(int $min, int $max): int {}
+
+    final class Random
+    {
+
+        private Random\NumberGenerator $randomNumberGenerator;
+
+        public function __construct(?Random\NumberGenerator $randomNumberGenerator = null) {}
+
+        public function getNumberGenerator(): Random\NumberGenerator {}
+
+        public function getInt(int $min, int $max): int {}
+
+        public function getBytes(int $length): string {}
+
+        public function shuffleArray(array $array): array {}
+
+        public function shuffleString(string $string): string {}
+
+        public function __serialize(): array {}
+
+        public function __unserialize(array $data): void {}
+
+    }
+
+}
+
+namespace Random
+{
+
+    abstract class NumberGenerator
```

Comment:
What do you think about making `NumberGenerator` an interface instead, since it doesn't have any behavior? It would be useful IMO, since userland implementations could extend whatever they want.

+ I'd put the interface into the `Random\NumberGenerator` namespace.